lib/actions/user.js

```diff
@@ -109,6 +133,88 @@ export function fetchAuth0Token(auth0, intl) {
   }
 }
 
+/**
+ * Converts a middleware trip request to the search format.
+ */
+function convertRequestToSearch(config) {
+  return function (tripRequest) {
+    const { dateCreated, id, requestParameters = {} } = tripRequest
+    const { host, path } = config.api
+    return {
+      canDelete: false,
+      id,
+      query: planParamsToQuery(requestParameters),
+      timestamp: dateCreated,
+      url: `${host}${path}/plan?${qs.stringify(requestParameters)}`
+    }
+  }
+}
+
+/**
+ * Removes duplicate requests saved from batch queries,
+ * so that only one request is displayed per batch.
+ * (Except for the mode, all query params in the same batch are the same.)
+ */
+function removeDuplicateRequestsFromBatch(config) {
+  if (isBatchRoutingEnabled(config)) {
+    const batches = {}
+    return function ({ query }) {
+      if (!batches[query.batchId]) {
+        batches[query.batchId] = true
+
+        // Remove the mode for display purposes
+        query.mode = ''
+        return true
+      }
+      return false
+    }
+  }
+  return (tripReq) => tripReq
+}
+
+/**
+ * Fetches the most recent (default 10) trip history for a user.
+ */
+export function fetchTripRequests() {
+  return async function (dispatch, getState) {
+    const { accessToken, apiBaseUrl, apiKey, loggedInUser } =
+      getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_TRIP_HISTORY_PATH}?userId=${loggedInUser.id}`
+
+    const { data: requests, status: emptyStatus } = await secureFetch(
+      `${requestUrl}&limit=0`,
+      accessToken,
+      apiKey,
+      'GET'
+    )
+
+    // FIXME: Revert to 10 after taking care of duplicates in the middleware.
+    const DEFAULT_LIMIT = 50
```

Comment:
Are the duplicates fixed?

Reply:
I think we would have to review how trip requests are recorded.